Compiling yuzu-mainline fails on latest builds [Linux]

Compiling yuzu-mainline under linux-gcc fails with commit 8f548266cd0493a537ed7edf56c869bd7420953a.  Last successful compile was several days before, on commit d551c910bbd0570e953b3c769e08ae7edc495e3c.  Can perform a bisect if needed.

**Update**: Performed a bisect.  First bad commit is b52097804334307103cfaba72cd6e27db804f650

Test system is Linux Mint 19.3 (ubuntu 18.04 based), kernel 5.3 with gcc 9.3.0.

Compile log excerpt:
```
/home/shoegzer/Desktop/yuzu/yuzu-mainline/src/common/lz4_compression.cpp:14:31: error: ‘LZ4_MAX_INPUT_SIZE’ was not declared in this scope
   14 |     ASSERT_MSG(source_size <= LZ4_MAX_INPUT_SIZE, "Source size exceeds LZ4 maximum input size");
      |                               ^~~~~~~~~~~~~~~~~~
/home/shoegzer/Desktop/yuzu/yuzu-mainline/src/./common/assert.h:36:11: note: in definition of macro ‘ASSERT_MSG’
   36 |     if (!(_a_)) {                                                                                  \
      |           ^~~
...
[ 54%] Linking CXX static library libdynarmic.a
[ 54%] Built target dynarmic
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```
Full log is here: [yuzu_compile.log](https://github.com/yuzu-emu/yuzu/files/4541287/yuzu_compile.log)

I also have fail of compile 
`[ 55%] Building CXX object src/video_core/CMakeFiles/video_core.dir/textures/astc.cpp.o
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:196:23: error: ‘static_vector_options’ is not a member of ‘boost::container’; did you mean ‘vector_options’?
  196 |     boost::container::static_vector_options<
      |                       ^~~~~~~~~~~~~~~~~~~~~
      |                       vector_options
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:197:27: error: ‘inplace_alignment’ is not a member of ‘boost::container’
  197 |         boost::container::inplace_alignment<alignof(IntegerEncodedValue)>,
      |                           ^~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:197:73: error: wrong number of template arguments (3, should be 2)
  197 |         boost::container::inplace_alignment<alignof(IntegerEncodedValue)>,
      |                                                                         ^
In file included from /usr/include/boost/container/vector.hpp:26,
                 from /usr/include/boost/container/static_vector.hpp:25,
                 from /home/user/yuzu-mainline/src/video_core/textures/astc.cpp:23:
/usr/include/boost/container/container_fwd.hpp:106:7: note: provided for ‘template<class T, long unsigned int Capacity> class boost::container::static_vector’
  106 | class static_vector;
      |       ^~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:200:51: error: ‘IntegerEncodedVector’ has not been declared
  200 | static void DecodeTritBlock(InputBitStream& bits, IntegerEncodedVector& result, u32 nBitsPerValue) {
      |                                                   ^~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: In function ‘void DecodeTritBlock(InputBitStream&, int&, u32)’:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:252:43: error: request for member ‘emplace_back’ in ‘result’, which is of non-class type ‘int’
  252 |         IntegerEncodedValue& val = result.emplace_back(IntegerEncoding::Trit, nBitsPerValue);
      |                                           ^~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: At global scope:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:258:52: error: ‘IntegerEncodedVector’ has not been declared
  258 | static void DecodeQus32Block(InputBitStream& bits, IntegerEncodedVector& result,
      |                                                    ^~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: In function ‘void DecodeQus32Block(InputBitStream&, int&, u32)’:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:299:43: error: request for member ‘emplace_back’ in ‘result’, which is of non-class type ‘int’
  299 |         IntegerEncodedValue& val = result.emplace_back(IntegerEncoding::Qus32, nBitsPerValue);
      |                                           ^~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: At global scope:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:35: error: variable or field ‘DecodeIntegerSequence’ declared void
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                   ^~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:35: error: ‘IntegerEncodedVector’ was not declared in this scope; did you mean ‘IntegerEncodedValue’?
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                   ^~~~~~~~~~~~~~~~~~~~
      |                                   IntegerEncodedValue
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:57: error: ‘result’ was not declared in this scope
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                                         ^~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:79: error: expected primary-expression before ‘&’ token
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                                                               ^
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:81: error: ‘bits’ was not declared in this scope; did you mean ‘Bits’?
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                                                                 ^~~~
      |                                                                                 Bits
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:346:91: error: expected primary-expression before ‘maxRange’
  346 | static void DecodeIntegerSequence(IntegerEncodedVector& result, InputBitStream& bits, u32 maxRange,
      |                                                                                           ^~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:347:39: error: expected primary-expression before ‘nValues’
  347 |                                   u32 nValues) {
      |                                       ^~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: In function ‘void ASTCC::DecodeColorValues(u32*, u8*, const u32*, u32, u32)’:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:898:5: error: ‘IntegerEncodedVector’ was not declared in this scope; did you mean ‘IntegerEncodedValue’?
  898 |     IntegerEncodedVector decodedColorValues;
      |     ^~~~~~~~~~~~~~~~~~~~
      |     IntegerEncodedValue
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:901:27: error: ‘decodedColorValues’ was not declared in this scope; did you mean ‘DecodeColorValues’?
  901 |     DecodeIntegerSequence(decodedColorValues, colorStream, range, nValues);
      |                           ^~~~~~~~~~~~~~~~~~
      |                           DecodeColorValues
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:901:5: error: ‘DecodeIntegerSequence’ was not declared in this scope
  901 |     DecodeIntegerSequence(decodedColorValues, colorStream, range, nValues);
      |     ^~~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: At global scope:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1128:59: error: ‘IntegerEncodedVector’ does not name a type; did you mean ‘IntegerEncodedValue’?
 1128 | static void UnquantizeTexelWeights(u32 out[2][144], const IntegerEncodedVector& weights,
      |                                                           ^~~~~~~~~~~~~~~~~~~~
      |                                                           IntegerEncodedValue
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: In function ‘void ASTCC::UnquantizeTexelWeights(u32 (*)[144], const int&, const ASTCC::TexelWeightParams&, u32, u32)’:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1134:29: error: request for member ‘begin’ in ‘weights’, which is of non-class type ‘const int’
 1134 |     for (auto itr = weights.begin(); itr != weights.end(); ++itr) {
      |                             ^~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1134:53: error: request for member ‘end’ in ‘weights’, which is of non-class type ‘const int’
 1134 |     for (auto itr = weights.begin(); itr != weights.end(); ++itr) {
      |                                                     ^~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1140:32: error: request for member ‘end’ in ‘weights’, which is of non-class type ‘const int’
 1140 |             if (itr == weights.end()) {
      |                                ^~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: In function ‘void ASTCC::DecompressBlock(const u8*, u32, u32, u32*)’:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1626:5: error: ‘IntegerEncodedVector’ was not declared in this scope; did you mean ‘IntegerEncodedValue’?
 1626 |     IntegerEncodedVector texelWeightValues;
      |     ^~~~~~~~~~~~~~~~~~~~
      |     IntegerEncodedValue
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1630:27: error: ‘texelWeightValues’ was not declared in this scope; did you mean ‘texelWeightData’?
 1630 |     DecodeIntegerSequence(texelWeightValues, weightStream, weightParams.m_MaxWeight,
      |                           ^~~~~~~~~~~~~~~~~
      |                           texelWeightData
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1630:5: error: ‘DecodeIntegerSequence’ was not declared in this scope
 1630 |     DecodeIntegerSequence(texelWeightValues, weightStream, weightParams.m_MaxWeight,
      |     ^~~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp: At global scope:
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1128:13: warning: ‘void ASTCC::UnquantizeTexelWeights(u32 (*)[144], const int&, const ASTCC::TexelWeightParams&, u32, u32)’ defined but not used [-Wunused-function]
 1128 | static void UnquantizeTexelWeights(u32 out[2][144], const IntegerEncodedVector& weights,
      |             ^~~~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:1039:12: warning: ‘u32 ASTCC::UnquantizeTexelWeight(const IntegerEncodedValue&)’ defined but not used [-Wunused-function]
 1039 | static u32 UnquantizeTexelWeight(const IntegerEncodedValue& val) {
      |            ^~~~~~~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:258:13: warning: ‘void DecodeQus32Block(InputBitStream&, int&, u32)’ defined but not used [-Wunused-function]
  258 | static void DecodeQus32Block(InputBitStream& bits, IntegerEncodedVector& result,
      |             ^~~~~~~~~~~~~~~~
/home/user/yuzu-mainline/src/video_core/textures/astc.cpp:200:13: warning: ‘void DecodeTritBlock(InputBitStream&, int&, u32)’ defined but not used [-Wunused-function]
  200 | static void DecodeTritBlock(InputBitStream& bits, IntegerEncodedVector& result, u32 nBitsPerValue) {
      |             ^~~~~~~~~~~~~~~
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/build.make:1116: src/video_core/CMakeFiles/video_core.dir/textures/astc.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2179: src/video_core/CMakeFiles/video_core.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
`

At least in the last 3 or 4 tags...

```sh
In file included from /home/wagner/Downloads/yuzu-mainline/src/video_core/renderer_vulkan/fixed_pipeline_state.cpp:12:
/home/wagner/Downloads/yuzu-mainline/src/./video_core/renderer_vulkan/fixed_pipeline_state.h: In member function ‘void Vulkan::FixedPipelineState::VertexInput::SetBinding(std::size_t, bool, u32, u32)’:
/home/wagner/Downloads/yuzu-mainline/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:132:35: error: conversion from ‘u32’ {aka ‘unsigned int’} to ‘short unsigned int’ may change value [-Werror=conversion]
  132 |             binding.stride.Assign(stride);
      |                                   ^~~~~~
In file included from /home/wagner/Downloads/yuzu-mainline/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:10,
                 from /home/wagner/Downloads/yuzu-mainline/src/video_core/renderer_vulkan/fixed_pipeline_state.cpp:12:
/home/wagner/Downloads/yuzu-mainline/src/./common/bit_field.h: In instantiation of ‘constexpr void BitField<Position, Bits, T, EndianTag>::Assign(const T&) [with long unsigned int Position = 0; long unsigned int Bits = 1; T = short unsigned int; EndianTag = KeepTag]’:
/home/wagner/Downloads/yuzu-mainline/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:131:51:   required from here
/home/wagner/Downloads/yuzu-mainline/src/./common/bit_field.h:183:63: error: conversion from ‘int’ to ‘BitField<0, 1, short unsigned int>::StorageTypeWithEndian’ {aka ‘short unsigned int’} may change value [-Werror=conversion]
  183 |         storage = (static_cast<StorageType>(storage) & ~mask) | FormatValue(value);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/wagner/Downloads/yuzu-mainline/src/./common/bit_field.h: In instantiation of ‘constexpr void BitField<Position, Bits, T, EndianTag>::Assign(const T&) [with long unsigned int Position = 1; long unsigned int Bits = 12; T = short unsigned int; EndianTag = KeepTag]’:
/home/wagner/Downloads/yuzu-mainline/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:132:41:   required from here
/home/wagner/Downloads/yuzu-mainline/src/./common/bit_field.h:183:63: error: conversion from ‘int’ to ‘BitField<1, 12, short unsigned int>::StorageTypeWithEndian’ {aka ‘short unsigned int’} may change value [-Werror=conversion]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,6 +157,7 @@ macro(yuzu_find_packages)
         #"libzip            1.5         libzip/1.5.2@bincrafters/stable"
         "lz4               1.8         lz4/1.9.2"
         "nlohmann_json     3.7         nlohmann_json/3.7.3"
+    # we need to be careful as the version check might be broken https://github.com/xiph/opus/issues/110
         "opus              1.3         opus/1.3.1"
         "ZLIB              1.2         zlib/1.2.11"
         "zstd              1.4         zstd/1.4.4"

--- a/externals/find-modules/FindCatch2.cmake
+++ b/externals/find-modules/FindCatch2.cmake
@@ -8,11 +8,25 @@ find_path(Catch2_INCLUDE_DIR
   PATH_SUFFIXES catch2
 )
 
+if(Catch2_INCLUDE_DIR)
+  file(STRINGS "${Catch2_INCLUDE_DIR}/catch.hpp" _Catch2_version_lines
+    REGEX "#define[ \t]+CATCH_VERSION_(MAJOR|MINOR|PATCH)")
+  string(REGEX REPLACE ".*CATCH_VERSION_MAJOR +\([0-9]+\).*" "\\1" _Catch2_version_major "${_Catch2_version_lines}")
+  string(REGEX REPLACE ".*CATCH_VERSION_MINOR +\([0-9]+\).*" "\\1" _Catch2_version_minor "${_Catch2_version_lines}")
+  string(REGEX REPLACE ".*CATCH_VERSION_PATCH +\([0-9]+\).*" "\\1" _Catch2_version_patch "${_Catch2_version_lines}")
+  set(Catch2_VERSION "${_Catch2_version_major}.${_Catch2_version_minor}.${_Catch2_version_patch}")
+  unset(_Catch2_version_major)
+  unset(_Catch2_version_minor)
+  unset(_Catch2_version_patch)
+  unset(_Catch2_version_lines)
+endif()
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Catch2
   FOUND_VAR Catch2_FOUND
   REQUIRED_VARS
     Catch2_INCLUDE_DIR
+    Catch2_VERSION
   VERSION_VAR Catch2_VERSION
 )
 

--- a/externals/find-modules/FindLibzip.cmake
+++ b/externals/find-modules/FindLibzip.cmake
@@ -28,6 +28,16 @@ find_library(LIBZIP_LIBRARY
   "$ENV{LIB_DIR}/lib" "$ENV{LIB}" /usr/local/lib /usr/lib
 )
 
+if (LIBZIP_INCLUDE_DIR_ZIPCONF)
+  FILE(READ "${LIBZIP_INCLUDE_DIR_ZIPCONF}/zipconf.h" _LIBZIP_VERSION_CONTENTS)
+  if (_LIBZIP_VERSION_CONTENTS)
+    STRING(REGEX REPLACE ".*#define LIBZIP_VERSION \"([0-9.]+)\".*" "\\1" LIBZIP_VERSION "${_LIBZIP_VERSION_CONTENTS}")
+  endif()
+  unset(_LIBZIP_VERSION_CONTENTS)
+endif()
+
+set(LIBZIP_VERSION ${LIBZIP_VERSION} CACHE STRING "Version number of libzip")
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Libzip
   FOUND_VAR LIBZIP_FOUND
@@ -35,18 +45,9 @@ find_package_handle_standard_args(Libzip
     LIBZIP_LIBRARY
     LIBZIP_INCLUDE_DIR
     LIBZIP_INCLUDE_DIR_ZIPCONF
+    LIBZIP_VERSION
+  VERSION_VAR LIBZIP_VERSION
 )
-
-set(LIBZIP_VERSION 0)
-
-if (LIBZIP_INCLUDE_DIR_ZIPCONF)
-  FILE(READ "${LIBZIP_INCLUDE_DIR_ZIPCONF}/zipconf.h" _LIBZIP_VERSION_CONTENTS)
-  if (_LIBZIP_VERSION_CONTENTS)
-    STRING(REGEX REPLACE ".*#define LIBZIP_VERSION \"([0-9.]+)\".*" "\\1" LIBZIP_VERSION "${_LIBZIP_VERSION_CONTENTS}")
-  endif()
-endif()
-
-set(LIBZIP_VERSION ${LIBZIP_VERSION} CACHE STRING "Version number of libzip")
 
 if(LIBZIP_FOUND)
   set(LIBZIP_LIBRARIES ${LIBZIP_LIBRARY})
@@ -65,5 +66,7 @@ endif()
 
 mark_as_advanced(
     LIBZIP_INCLUDE_DIR
+    LIBZIP_INCLUDE_DIR_ZIPCONF
     LIBZIP_LIBRARY
+    LIBZIP_VERSION
 )

--- a/externals/find-modules/Findfmt.cmake
+++ b/externals/find-modules/Findfmt.cmake
@@ -13,12 +13,38 @@ find_library(fmt_LIBRARY
   PATHS ${PC_fmt_LIBRARY_DIRS} ${CONAN_LIB_DIRS_fmt}
 )
 
+if(fmt_INCLUDE_DIR)
+  set(_fmt_version_file "${fmt_INCLUDE_DIR}/core.h")
+  if(NOT EXISTS "${_fmt_version_file}")
+    set(_fmt_version_file "${fmt_INCLUDE_DIR}/format.h")
+  endif()
+  if(EXISTS "${_fmt_version_file}")
+    # parse "#define FMT_VERSION 60200" to 6.2.0
+    file(STRINGS "${_fmt_version_file}" fmt_VERSION_LINE
+      REGEX "^#define[ \t]+FMT_VERSION[ \t]+[0-9]+$")
+    string(REGEX REPLACE "^#define[ \t]+FMT_VERSION[ \t]+([0-9]+)$"
+      "\\1" fmt_VERSION "${fmt_VERSION_LINE}")
+    foreach(ver "fmt_VERSION_PATCH" "fmt_VERSION_MINOR" "fmt_VERSION_MAJOR")
+      math(EXPR ${ver} "${fmt_VERSION} % 100")
+      math(EXPR fmt_VERSION "(${fmt_VERSION} - ${${ver}}) / 100")
+    endforeach()
+    set(fmt_VERSION
+      "${fmt_VERSION_MAJOR}.${fmt_VERSION_MINOR}.${fmt_VERSION_PATCH}")
+  endif()
+  unset(_fmt_version_file)
+  unset(fmt_VERSION_LINE)
+  unset(fmt_VERSION_MAJOR)
+  unset(fmt_VERSION_MINOR)
+  unset(fmt_VERSION_PATCH)
+endif()
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(fmt
   FOUND_VAR fmt_FOUND
   REQUIRED_VARS
     fmt_LIBRARY
     fmt_INCLUDE_DIR
+    fmt_VERSION
   VERSION_VAR fmt_VERSION
 )
 

--- a/externals/find-modules/Findnlohmann_json.cmake
+++ b/externals/find-modules/Findnlohmann_json.cmake
@@ -8,11 +8,25 @@ find_path(nlohmann_json_INCLUDE_DIR
   PATH_SUFFIXES nlohmann
 )
 
+if(nlohmann_json_INCLUDE_DIR)
+  file(STRINGS "${nlohmann_json_INCLUDE_DIR}/json.hpp" _nlohmann_json_version_lines
+    REGEX "#define[ \t]+NLOHMANN_JSON_VERSION_(MAJOR|MINOR|PATCH)")
+  string(REGEX REPLACE ".*NLOHMANN_JSON_VERSION_MAJOR +\([0-9]+\).*" "\\1" _nlohmann_json_version_major "${_nlohmann_json_version_lines}")
+  string(REGEX REPLACE ".*NLOHMANN_JSON_VERSION_MINOR +\([0-9]+\).*" "\\1" _nlohmann_json_version_minor "${_nlohmann_json_version_lines}")
+  string(REGEX REPLACE ".*NLOHMANN_JSON_VERSION_PATCH +\([0-9]+\).*" "\\1" _nlohmann_json_version_patch "${_nlohmann_json_version_lines}")
+  set(nlohmann_json_VERSION "${_nlohmann_json_version_major}.${_nlohmann_json_version_minor}.${_nlohmann_json_version_patch}")
+  unset(_nlohmann_json_version_major)
+  unset(_nlohmann_json_version_minor)
+  unset(_nlohmann_json_version_patch)
+  unset(_nlohmann_json_version_lines)
+endif()
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(nlohmann_json
   FOUND_VAR nlohmann_json_FOUND
   REQUIRED_VARS
     nlohmann_json_INCLUDE_DIR
+    nlohmann_json_VERSION
   VERSION_VAR nlohmann_json_VERSION
 )
 

--- a/externals/find-modules/Findzstd.cmake
+++ b/externals/find-modules/Findzstd.cmake
@@ -11,12 +11,26 @@ find_library(zstd_LIBRARY
   PATHS ${PC_zstd_LIBRARY_DIRS}
 )
 
+if(zstd_INCLUDE_DIR)
+  file(STRINGS "${zstd_INCLUDE_DIR}/zstd.h" _zstd_version_lines
+    REGEX "#define[ \t]+ZSTD_VERSION_(MAJOR|MINOR|RELEASE)")
+  string(REGEX REPLACE ".*ZSTD_VERSION_MAJOR *\([0-9]*\).*" "\\1" _zstd_version_major "${_zstd_version_lines}")
+  string(REGEX REPLACE ".*ZSTD_VERSION_MINOR *\([0-9]*\).*" "\\1" _zstd_version_minor "${_zstd_version_lines}")
+  string(REGEX REPLACE ".*ZSTD_VERSION_RELEASE *\([0-9]*\).*" "\\1" _zstd_version_release "${_zstd_version_lines}")
+  set(zstd_VERSION "${_zstd_version_major}.${_zstd_version_minor}.${_zstd_version_release}")
+  unset(_zstd_version_major)
+  unset(_zstd_version_minor)
+  unset(_zstd_version_release)
+  unset(_zstd_version_lines)
+endif()
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(zstd
   FOUND_VAR zstd_FOUND
   REQUIRED_VARS
     zstd_LIBRARY
     zstd_INCLUDE_DIR
+    zstd_VERSION
   VERSION_VAR zstd_VERSION
 )
 

